src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -147,7 +147,7 @@ public static DetailAST parse(FileContents contents)
                     "%s occurred while parsing file %s.",
                     ex.getClass().getSimpleName(), fileName);
             throw new CheckstyleException(exceptionMsg, ex);
-        };
+        }
 
         return compilationUnit;
```

Comment:
```suggestion
        throw new IllegalStateException("should not occure")
```
As we could make the return immediately now, we focus on where the action happens and avoid unnecessarily disrupting the code flow. Since we only have one valid case now, we should highlight this clearly.